crates/solver/src/settlement_submission/submitter/public_mempool_api.rs

```diff
@@ -119,6 +130,77 @@ impl TransactionSubmitting for PublicMempoolApi {
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum SubmissionNode {
+    /// Transactions that are sent to this nodes are expected to be broadcast to the mempool and
+    /// eventually be included in a block.
+    Broadcast(Web3),
+    /// A notification node is an endpoint that is not expected to submit transactions to the
+    /// mempool once a transaction has been received. Its purpose is notifying the node owner that a
+    /// transaction has been submitted.
+    /// In general, there are lower expectations on the availability of this node variant.
+    Notification(Web3),
+}
+
+impl Deref for SubmissionNode {
```

Comment:
Nice, I was reviewing last night and asking myself: “where is the code change that unwraps the inner web3 instance in the submission logic?”. The answer is “Rust magic!”.

Reply:
This is a bit of an anti pattern in Rust imo. I'd rather make this a normal function or make users of SubmissionNode access the web3 field directly.

Reply:
The docs do say "Deref should only be implemented for smart pointers to avoid confusion" so it's a bit eccessive.
Now that `SubmissionNode` is a struct, the unwrapping is just `node.web3` so I decided to remove `Deref` from the code.